cloud/azure/actuators/machine/machineactuator.go

```diff
@@ -197,18 +203,70 @@ func (azure *AzureClient) GetKubeConfig(cluster *clusterv1.Cluster, machine *clu
 	if err != nil {
 		return "", err
 	}
-	//az vm run-command invoke --name [vm_name] --resource-group [rg_name] --command-id RunShellScript --scripts 'sudo cat /etc/kubernetes/admin.conf'
-	script := "sudo cat /etc/kubernetes/admin.conf"
-	result := util.ExecCommand(
-		"az", "vm", "run-command", "invoke",
-		"--name", machine.ObjectMeta.Name,
-		"--resource-group", clusterConfig.ResourceGroup,
-		"--command-id", "RunShellScript",
-		"--scripts", script)
-	var parsed map[string]interface{}
-	err = json.Unmarshal([]byte(result), &parsed)
-	message := parsed["output"].([]map[string]interface{})[0]["message"].(string)
-	return message, nil
+	var machineConfig azureconfigv1.AzureMachineProviderConfig
+	err = azure.decodeMachineProviderConfig(machine.Spec.ProviderConfig, &machineConfig)
+	if err != nil {
+		return "", err
+	}
+
+	decoded, err := base64.StdEncoding.DecodeString(machineConfig.SSHPrivateKey)
+	privateKey := string(decoded)
+	if err != nil {
+		return "", err
+	}
+
+	ip, err := azure.GetIP(cluster, machine)
+	if err != nil {
+		return "", err
+	}
+	sshclient, err := GetSshClient(ip, privateKey)
+	if err != nil {
+		return "", fmt.Errorf("unable to get ssh client: %v", err)
+	}
+	sftpClient, err := sftp.NewClient(sshclient)
+	if err != nil {
+		return "", fmt.Errorf("Error setting sftp client: %s", err)
+	}
+
+	remoteFile := "/home/ClusterAPI/.kube/config"
```

Comment:
Can you make this parameterized with the `SSHUser` constant?

Reply:
Done